### PR TITLE
Prepare v0.45.0

### LINKED ISSUE #4871: Prepare core release v0.45.0

### PR DESCRIPTION
**Pull request by jpkrohling**
Contrib PR: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/7937

Closes #4871

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
codecov failing because there's 0% of diff hit... Seriously, codecov? :-)

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,16 +2,20 @@
 
 ## Unreleased
 
+## v0.45.0 Beta
+
 ### 🛑 Breaking changes 🛑
 
 - Remove deprecated funcs in configtelemetry (#4808)
 - Deprecate `service/defaultcomponents` go package (#4622)
+- `otlphttp` and `otlp` exporters enable gzip compression by default (#4632)
```
Comment (tigrannajaryan) [inline CHANGELOG.md:11]: Looks like this was incorrectly added to 0.44.0, right?

File: cmd/otelcorecol/builder-config.yaml [hunk 2]
```diff
@@ -2,29 +2,29 @@ dist:
   module: go.opentelemetry.io/collector/cmd/otelcorecol
   name: otelcorecol
   description: Local OpenTelemetry Collector binary, testing only.
-  version: 0.43.1-dev
```
Comment (tigrannajaryan) [inline cmd/otelcorecol/builder-config.yaml:5]: What does this version affect? Did we incorrectly release 0.44.0 without updating this field?

**Review by tigrannajaryan — APPROVED** [review]
LGTM, just a couple questions, but can be merged.

File: CHANGELOG.md (hunk 1 above)
Comment (jpkrohling) [reply CHANGELOG.md:11]: Correct, I moved them here to reflect the right version they got in.

File: cmd/otelcorecol/builder-config.yaml (hunk 2 above)
Comment (jpkrohling) [reply cmd/otelcorecol/builder-config.yaml:5]: No, this is what's displayed in the logs when you run `make run`. This distribution is only used for local debugging. The other versions on this file also don't matter that much, as the replaces statement down there points to the file system location for the modules.

**Pull request merged**